muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/tabs.tsx

```diff
@@ -28,58 +29,64 @@ interface TabsProps {
 }
 
 interface MobileOnlyTabsProps {
-  onTabChange: (id: string) => any,
+  onTabChange: (id: string) => void,
   activeTab: string,
   modifier?: string,
   tabs: Array<TabType>,
   renderAllComponents?: boolean
-  children?: React.ReactNode;
 }
 
-interface TabsState {
-}
+export const Tabs: React.FC<TabsProps> = (props) => {
 
-class Tabs extends React.Component<TabsProps, TabsState>{
-  render() {
-    const a11yConfig = {
-      enabled: true,
-    }
+  const { modifier, renderAllComponents, activeTab, onTabChange, tabs, children, allTabs } = props;
 
-    const paginationConfig = {
-      el: ".tabs__pagination-container",
-      modifierClass: "tabs__pagination-container--"
-    }
+  const a11yConfig = {
+    enabled: true,
+  }
 
-    return <div className={`tabs ${this.props.modifier ? "tabs--" + this.props.modifier : ""}`}>
-      <div className={`tabs__tab-labels ${this.props.modifier ? "tabs__tab-labels--" + this.props.modifier : ""}`}>
-        {this.props.tabs.map((tab, index) => {
-          return <div className={`tabs__tab ${this.props.modifier ? "tabs__tab--" + this.props.modifier : ""} ${tab.type ? "tabs__tab--" + tab.type : ""} ${tab.id === this.props.activeTab ? "active" : ""}`}
-            key={tab.id} onClick={this.props.onTabChange.bind(this, tab.id)}>{tab.name}</div>
-        })}
-        {this.props.children}
-      </div>
-      <div className="tabs__tab-data-container">
-        {this.props.tabs.filter(t => this.props.renderAllComponents || t.id === this.props.activeTab)
-          .map(t => <div key={t.id} className={`tabs__tab-data ${t.type ? "tabs__tab-data--" + t.type : ""}  ${t.id === this.props.activeTab ? "active" : ""}`}>
-            {t.component()}
-          </div>)}
-      </div>
-      <Swiper modules={[A11y, Pagination]} a11y={a11yConfig} pagination={paginationConfig} className="tabs__tab-data-container tabs__tab-data-container--mobile">
-        {this.props.tabs.map(t =>
-          <SwiperSlide key={t.id} >
-            <div className="tabs__mobile-tab">
-              <div className="tabs__pagination-container"> </div>
-              <div>{t.name}</div>
-              {t.mobileAction ? t.mobileAction : <div className="tabs__mobile-tab-spacer" />}
-            </div>
-            {t.component()}
-          </SwiperSlide>)}
-      </Swiper>
-    </div>
+  const paginationConfig = {
+    el: ".tabs__pagination-container",
+    modifierClass: "tabs__pagination-container--"
   }
+
+  let nextSlide = allTabs[allTabs.indexOf(activeTab) + 1];
+  let prevSlide = allTabs[allTabs.indexOf(activeTab) - 1];
```

Comment:
No need to use `let`as variable is not changed before actual using. Use `const` instead.